src/elements/ngm-dashboard.ts

```diff
@@ -42,13 +51,26 @@ export class NgmDashboard extends LitElementI18n {
   @state() projects: DashboardProject[] | undefined;
   @state() selectedViewIndx: number | undefined;
   private viewer: Viewer | null = null;
+  private assetConfigs: any = {};
+  private assets: Config[] | undefined;
 
   constructor() {
     super();
-    DashboardStore.viewIndex.subscribe(viewIndex => {
-      this.selectView(viewIndex);
+    DashboardStore.viewIndex.subscribe(async viewIndex => {
+      await this.selectView(viewIndex);
     });
     MainStore.viewer.subscribe(viewer => this.viewer = viewer);
+    MainStore.layersRemoved.subscribe(async () => {
+      if (this.selectedViewIndx !== undefined && this.assets) {
+        await Promise.all(this.assets.map(async layer => {
+          const data = await layer.promise;
+          data.show = true;
+          await (<any>document.getElementsByTagName('ngm-side-bar')[0]).onCatalogLayerClicked({
+            detail: {layer}
+          });
```

Comment:
It does not look good to simulate click on something of a distance component.
Is there a cleaner way to do it?

Reply:
The function is badly named because it's not a click but a custom event.  We have the exact same code here:
https://github.com/swissgeol/ngm/blob/master/src/layers/ngm-layers-upload.ts#L56-L60

This to use events ?

Reply:
What about this: https://github.com/swissgeol/ngm/commit/ed58de26c0d04ef39d3c59fed27fc86ad13b0df6 ?
Use CustomEvent instead of function call. The event name should be changed to something else (any name proposition ?)

Reply:
I think it is better to use rxjs here. I thought about it yesterday when made my updates. I can update this too

Reply:
oh, I didn't notice your commit, Fred. CustomEvent looks good too but maybe it should be layeradd instead of click

Reply:
but in this case, we lost await, and probably layers will be added incorrectly. Need to test

Reply:
It works fine

Reply:
I've pushed my commit to this branch
@vladyslav-tk: I also think rxjs is the way to go

Reply:
Thanks for the improvement. Looks more "webcomponent" now.